src/Analysis/Ast/Impl/Analyzer/Symbols/SymbolCollector.cs

```diff
@@ -229,6 +242,39 @@ internal sealed class SymbolCollector : PythonWalker {
                     DiagnosticSource.Analysis));
         }
 
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="fd"></param>
+        /// <returns></returns>
+        private bool IsRedefinedByDecorator(FunctionDefinition fd) {
+            if (fd.Decorators != null) {
+                foreach (var d in fd.Decorators.Decorators) {
+                    switch (d) {
+                        case NameExpression n:
+                            if (fd.Name.Equals(n.Name)) {
+                                return true;
+                            }
+                            break;
+                        case MemberExpression m:
+                            // pylint only compares one level member expressions for decorators
+                            // e.g 
+                            // @foo.a
+                            // def foo(self): ...
+                            // is valid but not
+                            // @foo.a.b
+                            // def foo(self): ... 
+                            var ne = m.Target as NameExpression;
+                            if (fd.Name.Equals(ne?.Name ?? string.Empty)) {
+                                return true;
```

Comment:
You do can compare to null, `Equals` won't crash. But `fd.Name` can be null, see above